src/buildVariables.ts

```diff
@@ -316,37 +385,32 @@ const buildCreateVariables = (introspectionResults: IntrospectionResult) => (
       }
 
       if (isObject(params.data[key])) {
-        const fieldsToConnect = buildReferenceField({
-          inputArg: params.data[key],
+        const inputType = findInputFieldForType(
           introspectionResults,
-          typeName: `${resource.type.name}CreateInput`,
-          field: key,
-          mutationType: PRISMA_CONNECT
-        });
+          `${resource.type.name}UpdateInput`,
+          key
+        );
 
-        // If no fields in the object are valid, continue
-        if (Object.keys(fieldsToConnect).length === 0) {
+        if (!inputType) {
           return acc;
         }
 
-        // Else, connect the nodes
-        return {
-          ...acc,
-          data: {
-            ...acc.data,
-            [key]: { [PRISMA_CONNECT]: { ...fieldsToConnect } }
-          }
-        };
-      }
-
-      // Put id field in a where object
-      if (key === 'id' && params.data[key]) {
-        return {
-          ...acc,
-          where: {
-            id: params.data[key]
-          }
-        };
+        if (inputType.kind !== 'SCALAR') {
+          const typeName = `${resource.type.name}CreateInput`;
+          const data = buildObjectMutationData({
+            inputArg: params.data[key],
+            introspectionResults,
+            typeName,
+            key
+          });
+          return {
+            ...acc,
+            data: {
+              ...acc.data,
+              ...data
+            }
+          };
+        }
```

Comment:
this part is mostly duplicated from buildUpdateVariables, we should generalize that later